### ISSUE TITLE
FP16 gives NaN loss when using pre-trained model

### ISSUE THREAD

**Issue by bluesky314** [pytorch/pytorch#43556]
I tried the new fp16 in native torch. However, when I continue my model training for my segmentation task I get loss as NaNs. With the same script, if I initialize the same model architecture from scratch then it works fine.
I set adam's 'ep' to 1e-4 as well but it made no difference. I get NaN loss from the first batch continuing my trained model. 

The model that I am continuing is already quite good at the task. I am guessing it is due to very low loss value but am not sure. How can I fix this?

Python 3.6 
Pytorch 1.6
Linux
pip installed lightning
CUDA Version: 10.1
Nvidia-V100

**Comment by mcarilli** [comment]
First you should check if your network fits any advanced use case in the [examples](https://pytorch.org/docs/stable/notes/amp_examples.html).

If you suspect the loss (or any other region) is the pain point, you can force the loss computation (or other region) to execute in FP32 following the last code sample in the [autocast docstring](https://pytorch.org/docs/stable/amp.html#torch.cuda.amp.autocast) (locally disable autocast and cast the region's inputs to FP32).

If you're confident your usage is correct, can you provide a minimal working example with synthetic data that produces nans with autocast but non-nans without autocast?

**Comment by daixiangzi** [comment]
same problem

**Comment by daixiangzi** [comment]
I think you should release your code. today, I meet same problem, but I find my loss function is error. when I use standard loss function ,loss is normal
